### PR TITLE
[CK API] 2d forward convolution composable kernel integration (multi-datatype)

### PR DESCRIPTION
**Pull request by iq136boy**
Integrated 2d forward convolution composable kernels into MIOpen.

### CODE REVIEW

**Comment by atamazov** [comment]
Please merge recent develop & resolve conflicts

**Review by JehandadKhan — APPROVED** [review]

**Review by junliume — APPROVED** [review]

**Comment by atamazov** [comment]
@averinevg @DrizztDoUrden ⚠️ We somehow missed this PR. However, we should review every PR that changes solver.hpp. Could you please review this change as soon as possible?

File: src/include/miopen/solver.hpp [hunk 1]
```diff
@@ -4444,6 +4444,10 @@ struct PerformanceConfigHipImplicitGemmFwdXdlops
         f(s.kernel_id, "kernel_id");
     }
     bool operator==(const PerformanceConfigHipImplicitGemmFwdXdlops& other) const;
+    template <typename DataType>
+    void Init(const ConvolutionContext& ctx);
+    template <typename DataType>
+    bool CheckIsSupportCKArgs(const ConvolutionContext& ctx) const;
```
Comment (averinevg) [inline src/include/miopen/solver.hpp:4450]: Please, make these functions `private`. They are not part of  the public API.

File: src/include/miopen/solver.hpp [hunk 2]
```diff
@@ -4478,6 +4482,13 @@ struct ConvHipImplicitGemmFwdXdlops final
     // we do expect that CK is faster than Naive), therefore we use a
     // value bigger than 0.01f, e.g. 0.02f.
     float GetWti(const ConvolutionContext&) const override { return 0.02f; };
+    template <typename DataType>
+    bool CheckCKApplicability(const ConvolutionContext& ctx) const;
+    template <typename DataType>
+    void RunCKSolution(const Handle& handle,
+                       const AnyInvokeParams& primitive_parameters,
+                       const ConvolutionContext& ctx,
+                       const PerformanceConfigHipImplicitGemmFwdXdlops& config) const;
```
Comment (averinevg) [inline src/include/miopen/solver.hpp:4491]: Please, make these functions `private`. They are not part of  the public API.

File: src/solver/conv_hip_implicit_gemm_fwd_xdlops.cpp [hunk 3]
```diff
@@ -85,46 +90,161 @@ struct CKArgs
...
+        args.N,
+        args.K,
+        args.C,
+        args.input,
+        args.filter,
+        args.output,
+        args.strides,
+        args.dilation,
+        args.lPadding,
+        args.rPadding,
```
Comment (atamazov) [inline src/solver/conv_hip_implicit_gemm_fwd_xdlops.cpp:212]: [Informative] That's what #1958 is about. Missed during review.

**Comment by atamazov** [comment]
> @averinevg @DrizztDoUrden ⚠️ We somehow missed this PR. However, we should review every PR that changes solver.hpp...

/cc @junliume @JehandadKhan

**Comment by junliume** [comment]
@iq136boy  could you follow the review comments above and make necessary changes in following up PRs? Thanks!

**Pull request merged**